targets/target-mbed/source/jerry_port.c

```diff
@@ -0,0 +1,141 @@
+ * limitations under the License.
+ */
+
+#define _BSD_SOURCE
+#include <stdarg.h>
+#include <stdlib.h>
+#include <sys/time.h>
+
+#include "jerry-core/jerry-port.h"
+
+#include "mbed-hal/us_ticker_api.h"
+
+static bool abort_on_fail = false;
+
+/**
+ * Provide log message to filestream implementation for the engine.
+ */
+int jerry_port_logmsg (FILE *stream, /**< stream pointer */
+                       const char *format, /**< format string */
+                       ...) /**< parameters */
+{
+  va_list args;
+  int count;
+  va_start (args, format);
+  count = vfprintf (stream, format, args);
+  va_end (args);
+  return count;
+} /* jerry_port_logmsg */
+
+/**
+ * Provide error message to console implementation for the engine.
+ */
+int jerry_port_errormsg (const char *format, /**< format string */
+                         ...) /**< parameters */
+{
+  va_list args;
+  int count;
+  va_start (args, format);
+  count = vfprintf (stderr, format, args);
+  va_end (args);
+  return count;
+} /* jerry_port_errormsg */
+
+/**
+ * Provide output character to console implementation for the engine.
+ */
+int jerry_port_putchar (int c) /**< character to put */
+{
+  return putchar ((unsigned char) c);
+} /* jerry_port_putchar */
+
+/**
+ * Sets whether 'abort' should be called instead of 'exit' upon exiting with
+ * non-zero exit code in the default implementation of jerry_port_fatal.
+ */
+void jerry_port_default_set_abort_on_fail (bool flag) /**< new value of 'abort on fail' flag */
+{
+  abort_on_fail = flag;
+} /* jerry_port_default_set_abort_on_fail */
+
+/**
+ * Check whether 'abort' should be called instead of 'exit' upon exiting with
+ * non-zero exit code in the default implementation of jerry_port_fatal.
+ *
+ * @return true - if 'abort on fail' flag is set,
+ *         false - otherwise.
+ */
+bool jerry_port_default_is_abort_on_fail ()
+{
+  return abort_on_fail;
+} /* jerry_port_default_is_abort_on_fail */
+
+/**
+ * Default implementation of jerry_port_fatal.
+ */
+void jerry_port_fatal (jerry_fatal_code_t code)
+{
+  if (code != 0
+      && code != ERR_OUT_OF_MEMORY
+      && jerry_port_default_is_abort_on_fail ())
+  {
+    abort ();
+  }
+  else
+  {
+    exit (code);
+  }
+} /* jerry_port_fatal */
+
+/**
+ * Implementation of jerry_port_get_time_zone.
+ */
+bool jerry_port_get_time_zone (jerry_time_zone_t *tz_p)
+{
+  struct timeval tv;
+  struct timezone tz;
+
+  /* gettimeofday may not fill tz, so zero-initializing */
+  
+  tz.tz_minuteswest = 0;
+  tz.tz_dsttime = 0;
+
+  gettimeofday (&tv, &tz);
+
+  tz_p->offset = tz.tz_minuteswest;
+  tz_p->daylight_saving_time = tz.tz_dsttime > 0 ? 1 : 0;
+
+  return true;
+} /* jerry_port_get_time_zone */
+
+/**
+ * Implementation of jerry_port_get_current_time.
+ */
+double jerry_port_get_current_time ()
+{
+  struct timeval tv;
+
+  gettimeofday (&tv, NULL);
+
+  return ((double) tv.tv_sec) * 1000.0 + ((double) tv.tv_usec) / 1000.0;
```

Comment:
Similarly here: you might not need `gettimeofday` but you can just move the logic from `_gettimeofday` here. Something like:
```c
return (double) us_ticker_read ();
```
(If I get milli-micro-nano conversions right...)